documentation/api/processes.md

```diff
@@ -0,0 +1,9 @@
+# Processes
+
+Operations for obtaining information about discovered processes.
```

Comment:
Is there a simpler way to describe this? Maybe something along the lines of:

> The Processes API enables enumeration of the processes that `dotnet-monitor` can connect to and obtaining their metadata (such as environment variables and process names).

Should we consider a brief note describing the whole _'processes can change id between calls if they restart while you aren't looking'_ thing?

Reply:
... and possibly call out what we can connect to?

> `dotnet-monitor` can only connect to .NET Core 3.1, or .NET 5+ applications. When connecting to .NET Core 3.1 applications, some features maybe unavailable.

Reply:
I'm trying to think of a good place to add the second part rather than placing it on every page in the docs. Maybe a centralized doc area for this and then brief mentions with links to that doc.

I'd like to make these changes after @shirhatti merges everything together.